desktop/src/app/components/widgets/changes-history-modal.html

```diff
@@ -0,0 +1,61 @@
+<div class="modal-header">
+    <h5  class="modal-title" i18n="@@widgets.historyModal.header">
+        Bearbeitungshistorie der <b>{{document.resource.identifier}}</b>
+    </h5>
+
+</div>
+
+<div id="changes-history-modal-body" class="modal-body" >
+    <div class="resource-creation-info-row">
+        <span class="resource-creation-info-label" i18n="@@resource.creator.name">Erstellt von: </span>
+        <span class="resource-creation-info-text">{{document.created.user | slice:0:32}}</span>
+    </div>    
+    <div class="resource-creation-info-row">
+        <span class="resource-creation-info-label" i18n="@@resource.creation.date">Erstellungsdatum: </span>
+        <span class="resource-creation-info-text">{{formatDateTime(document.created.date)}}</span>
+    </div>
+    
+    
+
+    <div *ngIf="document.modified[0].user.length>=1; else noChangeTemplate">
+        <br />
+        <div id="list" class="col-lg-12 col-md-12 col-sm-12">               
+            <div id="list-header">  
+                <div class="list-header-field text-right"
+                    i18n="@@changes.list.colnames.date" 
+                    (click)="toggleColumnSort('date')" 
+                    [ngClass]="toggledDate ? 'arrow-down' : 'arrow-up'">
+                    <span class="box-left">Änderungsdatum </span>
+                </div>
+                <div class="list-header-field text-right" 
+                    i18n="@@changes.list.colnames.user"
+                    (click)="toggleColumnSort('user')"
+                    [ngClass]="toggledUser ? 'arrow-down' : 'arrow-up'">
+                    <span class="box-left">Benutzer </span>
+                </div>
+            </div>            
+            <div class="list-body">
+                <div *ngFor="let modification of document.modified"
+                    class = "list-row">                    
+                    <div class= "list-body-field" [textContent]="formatDateTime(modification.date)"></div>
+                    <div class= "list-body-field" [textContent]="modification.user.slice(0,32)"></div>      
+                </div> 
+            </div>
+        </div>
+    </div>  
+
+    <div #noChangeTemplate [hidden]="document.modified[0].user.length>=1">
+        <div id ="no-change-div" i18n="@@text.when.no.change">
+            <br />
+            Es wurde noch keine Änderung für die Ressource aufgeführt.    
+        </div>
+    </div>
+</div>   
+<br />
+
+<div class="modal-footer">
+    <button id="overview-save-confirmation-modal-cancel-button" type="button" class="btn btn-secondary"
+            (click)="activeModal.dismiss('cancel')" i18n="@@buttons.cancel">Abbrechen</button>
```

Comment:
Cancel buttons should only be used for modals where you perform an action (e.g. move or delete a resource). In this case, as the user you're not really cancelling anything, you're just closing the modal as expected. Therefore, in this case it's better to use a circle shaped close button in the modal header (like it is used e. g. in the QR code editor modal). You can remove the modal footer then.

Reply:
This is because the menu service is not yet set when opening the modal from the docedit component.

When the escape button is pressed in the docedit component, the menu context is checked. If the context is DOCEDIT, then the modal is closed. So you need to set the context to HISTORY_MODAL before opening the modal (and reset it afterwards) like it's already done when opening the modal from the resources component. The initialize function call is btw also yet missing when opening the modal from docedit.